opnode/test/system_test.go

```diff
@@ -343,7 +343,66 @@ func TestSystemE2E(t *testing.T) {
 	seqBlock, err := l2Seq.BlockByNumber(context.Background(), receipt.BlockNumber)
 	require.Nil(t, err)
 	require.Equal(t, verifBlock.Hash(), seqBlock.Hash(), "Verifier and sequencer blocks not the same after including a batch tx")
+}
+
+func TestMintOnRevertedDeposit(t *testing.T) {
+	log.Root().SetHandler(log.DiscardHandler()) // Comment this out to see geth l1/l2 logs
+
+	cfg := defaultSystemConfig(t)
+
+	sys, err := cfg.start()
+	require.Nil(t, err, "Error starting up system")
+	defer sys.Close()
+
+	l1Client := sys.Clients["l1"]
+	l2Verif := sys.Clients["verifier"]
+
+	// Find deposit contract
+	depositContract, err := deposit.NewDeposit(cfg.DepositContractAddress, l1Client)
+	require.Nil(t, err)
+
+	// Initialize our sender address
+	ethPrivKey, err := sys.wallet.PrivateKey(accounts.Account{
+		URL: accounts.URL{
+			Path: transactorHDPath,
+		},
+	})
+	require.Nil(t, err)
+	opts, err := bind.NewKeyedTransactorWithChainID(ethPrivKey, cfg.L1ChainID)
+	fromAddr := crypto.PubkeyToAddress(ethPrivKey.PublicKey)
+	require.Nil(t, err)
+
+	toAddr := common.Address{0xff, 0xff}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 1*time.Second)
+	startBalance, err := l2Verif.BalanceAt(ctx, fromAddr, nil)
+	cancel()
+	require.Nil(t, err)
+	mintAmount := big.NewInt(9_000_000) // total balance is 10_000_000
+	opts.Value = mintAmount
+	value := mintAmount.Mul(mintAmount, common.Big2) // trigger a revert by transferring more than we have available
+	tx, err := depositContract.DepositTransaction(opts, toAddr, value, big.NewInt(1_000_000), false, nil)
+	require.Nil(t, err, "with deposit tx")
 
+	receipt, err := waitForTransaction(tx.Hash(), l1Client, 6*time.Second)
+	require.Nil(t, err, "Waiting for deposit tx on L1")
+
+	reconstructedDep, err := derive.UnmarshalLogEvent(receipt.Logs[0])
+	require.NoError(t, err, "Could not reconstruct L2 Deposit")
+	tx = types.NewTx(reconstructedDep)
+	receipt, err = waitForTransaction(tx.Hash(), l2Verif, 6*time.Second)
+	require.NoError(t, err)
+	require.Equal(t, receipt.Status, types.ReceiptStatusSuccessful)
+
+	// Confirm balance
```

Comment:
Another check (which I think would fail right now) is to make sure that the nonce of the depositing account is bumped.

Reply:
No execution happened though, only one balance was increased (which you could think of as the system moving the funds), should we even increment the nonce of the account?

Reply:
I think so. Reverted transactions also bump the nonce. It's weird that some deposit txns increase the nonce but others do not.

Reply:
Indeed. This test should also check the nonce.

Reply:
Added that to the test, and implemented the nonce increment here: https://github.com/ethereum-optimism/reference-optimistic-geth/pull/10/commits/7058816e614c29c7f509f8e1ec1623ee154a22c5